### DISCUSSION TITLE
How to find an extension method using debug interfaces (IMetadata*, ICorDebug*  or any other) ?

### DISCUSSION THREAD

**Discussion by o-lek** [dotnet/runtime#74987] [Q&A]
We can find any method of a class using IMetaDataImport interface's methods EnumMethods(..) or EnumMethodsWithName(...).
But those methods don't enumerate [extension methods](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods) of a class. Is there any way to find those methods of a class or instance of a class ?

**Comment by davidfowl** [comment]
Extension methods are a C# concept not a runtime concept, these are just static methods on static classes marked with an [Extension] attribute where the first argument is the "this" pointer. They can be defined on any class in any assembly so there's no simple way to enumerate them at runtime. e.g.

https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoTA3CAnAAgGcCBeAgIgAsBTAGzoHtGCB3RvOgEwoG5MiAOgCyACgCU/LBgQBmAghQA2BUgIBhAgG9MBPQvmKVCACwExAFypQSAeWAArGgGMLBMOO279AX0w+gA===
